src/service/xhr-impl.js

```diff
@@ -576,6 +593,13 @@ export class FetchResponseHeaders {
  * @return {!Xhr}
  */
 export function xhrServiceForTesting(window) {
+  registerServiceBuilder(window, 'ampdoc', function() {
+    return {
+      isSingleDoc() {
+        return false;
+      },
+    };
+  });
```

Comment:
If we have to, we can even test for the presence of the service and `if` it out if not.